services/data-info/src/data_info/services/metadata.clj

```diff
@@ -181,25 +196,82 @@
 (defn metadata-set
   "Allows user to set metadata on an item with the given data-id. The user must exist in iRODS and have
    write permissions on the data item. The 'irods-avus' parameter should be an array of AVU maps following
-   the format used for (metadata-add)."
-  [user data-id irods-avus]
+   the format used for (metadata-add). The 'metadata' parameter should be in a format expected by the
+   metadata service set AVUs endpoint, or nil."
+  [user data-id {:keys [irods-avus metadata]}]
   (with-jargon (cfg/jargon-cfg) [cm]
     (validators/user-exists cm user)
     (let [{:keys [path type]} (uuids/path-for-uuid cm user data-id)
           irods-avus (set (map #(select-keys % [:attr :value :unit]) irods-avus))
           current-avus (set (list-path-metadata cm path :system false))
-          delete-irods-avus (s/difference current-avus irods-avus)]
+          delete-irods-avus (s/difference current-avus irods-avus)
+          metadata-request (json/encode (or metadata {}))]
       (validators/path-writeable cm user path)
       (authorized-avus irods-avus)
 
+      (metadata/set-avus user (resolve-data-type type) data-id metadata-request)
       (doseq [del-avu delete-irods-avus]
         (common-metadata-delete cm path del-avu))
       (doseq [avu irods-avus]
         (common-metadata-add cm path avu))
 
       {:path path
-       :user user
-       :type type})))
+       :user user})))
+
+(defn- find-attributes
```

Comment:
I think there's a more efficient way to do this with better primitives in clj-jargon but it might not matter for right now...